src/main/java/com/brandwatch/kafka_pod_autoscaler/triggers/KafkaThroughputTriggerProcessor.java

```diff
@@ -0,0 +1,131 @@
+package com.brandwatch.kafka_pod_autoscaler.triggers;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.google.auto.service.AutoService;
+
+import brandwatch.com.v1alpha1.KafkaPodAutoscaler;
+import brandwatch.com.v1alpha1.kafkapodautoscalerspec.Triggers;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import com.brandwatch.kafka_pod_autoscaler.ScaledResource;
+import com.brandwatch.kafka_pod_autoscaler.cache.KafkaMetadataCache;
+
+@Slf4j
+@AutoService(TriggerProcessor.class)
+public class KafkaThroughputTriggerProcessor implements TriggerProcessor {
+    private static final LoadingCache<TopicConsumerGroupId, Cache<PartitionTimestamp, Long>> consumerOffsetHistory = Caffeine
+        .newBuilder()
+        .expireAfterAccess(Duration.ofMinutes(5L))
+        .build(key -> Caffeine.newBuilder()
+            .expireAfterWrite(Duration.ofMinutes(10L))
+            .build());
+    private static final LoadingCache<TopicConsumerGroupId, Cache<PartitionTimestamp, Long>> topicEndHistory = Caffeine
+        .newBuilder()
+        .expireAfterAccess(Duration.ofMinutes(5L))
+        .build(key -> Caffeine.newBuilder()
+            .expireAfterWrite(Duration.ofMinutes(10L))
+            .build());
+
+    @Setter // (onMethod_ = { @VisibleForTesting })
```

Comment:
should the `onMethod...` be commented in here?

Reply:
Yeah, I didn't want to add guava to the proect just for this

Reply:
oh, right 😄